win32ss/user/user32/windows/cursoricon.c

```diff
@@ -2032,6 +2032,24 @@ HANDLE WINAPI CopyImage(
     TRACE("hImage=%p, uType=%u, cxDesired=%d, cyDesired=%d, fuFlags=%x\n",
         hImage, uType, cxDesired, cyDesired, fuFlags);
 
+    /* If hImage is NULL, then do fast return after setting LastError. */
+    if (!hImage)
+    {
+        switch(uType)
+        {
+            case IMAGE_BITMAP:
+                SetLastError(ERROR_INVALID_HANDLE);
+                break;
+            case IMAGE_CURSOR:
+            case IMAGE_ICON:
+                SetLastError(ERROR_INVALID_CURSOR_HANDLE);
+                break;
+            default:
+                SetLastError(ERROR_INVALID_PARAMETER);
+        }
+        return NULL;
+    }
+
     if (fuFlags & ~COPYIMAGE_VALID_FLAGS)
```

Comment:
Can you also test the following:
```c
CopyImage(NULL, IMAGE_BITMAP, 16, 16, 0x80000000);
```
(`0x80000000` is an invalid flag value), in order to test whether the flags validation is done first (we would get an `ERROR_INVALID_PARAMETER`), or whether the handle NULL check is done first instead, as you suggest in your PR (we would get an `ERROR_INVALID_HANDLE`).

Reply:
I added another test as you requested and it showed that with the ```0x80000000``` invalid flag value and the NULL handle, the LastError returned was ```ERROR_INVALID_PARAMETER```. Therefore, I moved the HANDLE test after the Invalid Flags test in the CopyImage function. That was a good idea, @HBelusca. Thanks.